cdc_vaccines/delphi_cdc_vaccines/pull.py

```diff
@@ -0,0 +1,136 @@
+                num_cols=df.shape[1],
+                min_date=min(df['Date']),
+                max_date=max(df['Date']),
+                checksum=hashlib.sha256(pd.util.hash_pandas_object(df).values).hexdigest())
+    df.columns = [i.lower() for i in df.columns]
+
+    df['recip_state'] = df['recip_state'].str.lower()
+    drop_columns.extend([x for x in df.columns if ("pct" in x) | ("svi" in x)])
+    drop_columns =  list(set(drop_columns))
+    df = GeoMapper().add_geocode(df, "state_id", "state_code",
+        from_col="recip_state", new_col="state_id", dropna=False)
+    df['state_id'] = df['state_id'].fillna('0').astype(int)
+    # Change FIPS from 0 to XX000 for statewise unallocated cases/deaths
+    unassigned_index = (df["fips"] == "UNK")
+    df.loc[unassigned_index, "fips"] = df["state_id"].loc[unassigned_index].values * 1000
+
+    # Conform FIPS
+    df["fips"] = df["fips"].apply(lambda x: f"{int(x):05d}")
+    df["timestamp"] = pd.to_datetime(df["date"])
+    # Drop unnecessary columns (state is pre-encoded in fips)
+    try:
+        df.drop(drop_columns, axis=1, inplace=True)
+    except KeyError as e:
+        raise ValueError(
+            "Tried to drop non-existent columns. The dataset "
+            "schema may have changed.  Please investigate and "
+            "amend drop_columns."
+        ) from e
+    # timestamp: str -> datetime
+    df.columns = ["fips",
+                  "cumulative_counts_tot_vaccine",
+                  "cumulative_counts_tot_vaccine_12P",
+                  "cumulative_counts_tot_vaccine_18P",
+                  "cumulative_counts_tot_vaccine_65P",
+                  "cumulative_counts_part_vaccine",
+                  "cumulative_counts_part_vaccine_12P",
+                  "cumulative_counts_part_vaccine_18P",
+                  "cumulative_counts_part_vaccine_65P",
+                  "timestamp"]
+    df_dummy = df.loc[(df["fips"]!='00000') & (df["timestamp"] == min(df["timestamp"]))].copy()
+    #handle fips 00000 separately
+    df_oth = df.loc[((df["fips"]=='00000') &
+        (df["timestamp"]==min(df[df['fips'] == '00000']['timestamp'])))].copy()
+    df_dummy = pd.concat([df_dummy, df_oth])
+    df_dummy.loc[:, "timestamp"] = df_dummy.loc[:, "timestamp"] - pd.Timedelta(days=1)
+    df_dummy.loc[:, ["cumulative_counts_tot_vaccine",
+                    "cumulative_counts_tot_vaccine_12P",
+                    "cumulative_counts_tot_vaccine_18P",
+                    "cumulative_counts_tot_vaccine_65P",
+                    "cumulative_counts_part_vaccine",
+                    "cumulative_counts_part_vaccine_12P",
+                    "cumulative_counts_part_vaccine_18P",
+                    "cumulative_counts_part_vaccine_65P",
+                    ]] = 0
+
+    df =pd.concat([df_dummy, df])
+    # Obtain new_counts
+    df.sort_values(["fips", "timestamp"], inplace=True)
+    for to, from_d in DIFFERENCE_MAPPING.items():
+        df[to] = df[from_d].diff()
```

Comment:
Btw, you might like [this](https://github.com/cmu-delphi/covidcast-indicators/blob/b4c9b5573232b89c45cf3ce8ea6d0bf61ef53133/jhu/delphi_jhu/pull.py#L27) version of taking diffs, grouped by geos.

Reply:
I think the `.isna()` in that code should probably be replaced with `.loc[min_time_value, :]` or something, but the gist is there.

Reply:
I wonder if we can keep this method for now, but then later I'll look to fix the method and use it.

Reply:
Sure, that's alright. A possible refactor for later.